tests/integration/components/power-select/type-ahead-test.js

```diff
@@ -0,0 +1,170 @@
+import { moduleForComponent, test } from 'ember-qunit';
+import hbs from 'htmlbars-inline-precompile';
+import { optionAtIndex } from 'ember-power-select/utils/group-utils';
+import { triggerKeydown, clickTrigger, typeInSearch } from 'ember-power-select/test-support/helpers';
+import { charCode, namesStartingWithA, numbers } from '../constants';
+import { find } from 'ember-native-dom-helpers';
+import { run } from '@ember/runloop';
+
+const WITH_EPS_CLOSED = {
+  beforeInteraction(trigger, assert) {
+    trigger.focus();
+    assert.equal(trigger.textContent.trim(), '', 'no value selected');
+    assert.notOk(find('.ember-power-select-dropdown'), 'The dropdown is closed');
+  },
+
+  checkSelectedValue(value, trigger, assert) {
+    // -1 adjust the position for the implicit placeholder
+    assert.equal(trigger.textContent.trim(), value);
+    assert.notOk(find('.ember-power-select-dropdown'), 'The dropdown is still closed');
+  },
+
+  valueAt(coll, _idx, isGrouped = false) {
+    let idx = isGrouped ? _idx - 1 : (_idx - 1) % coll.length;
+    return optionAtIndex(coll, idx).option;
+  }
+};
+
+const WITH_EPS_OPEN = {
+  beforeInteraction(trigger, assert) {
+    clickTrigger();
+    assert.ok(find('.ember-power-select-dropdown'), 'The dropdown is open');
+  },
+
+  checkSelectedValue(value, trigger, assert) {
+    assert.equal(find('.ember-power-select-option[aria-current=true]').textContent.trim(), value);
+    assert.ok(find('.ember-power-select-dropdown'), 'The dropdown is still open');
+  },
+
+  valueAt(coll, _idx, isGrouped = false) {
+    let idx = isGrouped ? _idx : _idx % coll.length;
+    return optionAtIndex(coll, idx).option;
+  }
+};
+
+function type(trigger, str, times = 1) {
```

Comment:
nit pick: Rename to `typeText` or `typeString`. Otherwise it seems a noun to me, synonym of "kind"

Reply:
Done.